### ISSUE TITLE
Changing `module.exports` to a reference causes crash/hang/shenanigans 

### ISSUE THREAD

**Issue by evantahler** [gotwarlost/istanbul#296]
I use Istanbul to test the [actionhero](http://www.actionherojs.com/) framework.   One of the features we have is "developmer mode" which will reload various files as you work on them so you don't have to restart the app.  We test this [in our test suite](https://github.com/evantahler/actionhero/blob/master/test/core/developerMode.js).

In [a recent PR](https://github.com/evantahler/actionhero/pull/545) we changed the way we exported from a file from a local reference:
```javascript
var thing = {};
exports.thing = thing;
```
to an inline declaration:
```javascript
exports.thing = {};
```
However, the test replaced this file in the old way.  When the underlying file changes the export pattern during the test suite (mocha) which is being analyzed, Istanbul crashes or hangs.  Here's [a travis build](https://travis-ci.org/evantahler/actionhero/builds/46251826) showing that off. 

I imagine you probably don't want to account for a crazy test like this, but I wanted to list it here if anyone comes across a similar issue.

**Comment by gotwarlost** [comment]
I don't think it is the reference change that is causing this. The explanation I can think of is that the coverage variable for the old and new  files are internally different because of the line number changes. The old object is being used for the new file. Not sure why that would cause a hang though. I would have expected an error to be thrown.
